### ISSUE TITLE
In Black Mesa (maybe other games as well) DXVK seems to allocate a lot more memory than D3D9

### ISSUE THREAD

**Issue by Deus-nsf** [doitsujin/dxvk#2585]
Booting to the game menu sees an additional +1.2 GB of RAM usage compared to D3D9 (after waiting for a few seconds), I'm using Windows to compare so maybe the memory management is different under Linux (I'll look at it too).
I tried with different versions from 1.9.0 to the latest and all of them add a good gigabyte of RAM allocation.

memory behavior, first launching the game in D3D9, then with DXVK:
![image](https://user-images.githubusercontent.com/13644590/161952294-bb149b2f-ab1f-40a0-8587-a50a79292ea6.png)

**Comment by Deus-nsf** [comment]
nooo, not the API trace hahaha
actually, you can totally use the one I did for the bush thing? it definitely captured the game being loaded, or do you want a fresh one?

**Comment by K0bin** [comment]
Your apitrace uses 80MB of additional memory compared to D3D9 on my system. Noticeable but far from the 1.2GB you're seeing here.

**Comment by Deus-nsf** [comment]
Bad news, during the API trace the memory usage was in line with the D3D9 behaviour, so I don't think it's going to expose anything relevant :/

here it is anyways: https://mega.nz/file/XgpHwT6Q#ibDikkdBIV33FKYTnKsRnFjxVJCTMlIgy2NjuY-bO7E

**Comment by K0bin** [comment]
> Bad news, during the API trace the memory usage was in line with the D3D9 behaviour, so I don't think it's going to expose anything relevant :/

That's normal, if you did it on Windows, it will use regular D3D9 to record the trace. The differences will hopefully show up when replaying the apitrace with DXVK.

**Comment by K0bin** [comment]
That apitrace doesn't even go into the game. Do you usually see that 1.2GB difference on the main menu?

**Comment by Deus-nsf** [comment]
yes, repro rate 100% (im switching to linux to see if the behaviour is the same)

**Comment by K0bin** [comment]
The main menu uses 500MB with Windows Nvidia D3D9 and 720MB with DXVK. Almost all of that difference comes down to how the DXVK memory allocator works and is not a problem.

Can you run it with DXVK_HUD=full and post a screenshot? (also make sure it's actually using more memory than expected when doing that)

**Comment by Deus-nsf** [comment]
Well, on linux, using the same game config + mods and the same dxvk.conf than my game build on windows, the problem isn't there, memory usage is normal, seems like it's a windows only thing, since dxvk doesn't officially support windows are you still interested if I look at this?

**Comment by K0bin** [comment]
I dont quite understand. Does the problem only happen with DXVK on Windows? Does it only happen with mods?

**Comment by Deus-nsf** [comment]
Mods and game config and dxvk.conf are the same between the 2 OSes so they are not responsible, the problem only occurs with dxvk on windows. The bush thing happens on linux too though.

**Comment by Deus-nsf** [comment]
![362890_20220406142945_1](https://user-images.githubusercontent.com/13644590/161974902-8d5806eb-1cc7-45e3-a442-96e352fef44e.png)

see, almost 2 GB in the menu! That's so weird! I'm going to try to update my GPU drivers just in case but they are pretty recent

**Comment by Deus-nsf** [comment]
I updated my GPU drivers, tried to disabled the nvidia shader cache in case it may be linked, even put the game in absolute lowest settings, still that same +-2 GB of RAM increase

**Comment by K0bin** [comment]
I still don't fully understand, does the memory increase only happen on Linux or on Windows?

**Comment by Deus-nsf** [comment]
only on windows and only with dxvk, i'm still trying to figure out what's happening!

**Comment by Deus-nsf** [comment]
You can close this issue, I found the culprit, it is the option "shader pre-caching" on steam under Windows, you want that disabled at all costs LOL.
